object.c

```diff
@@ -437,20 +424,17 @@ freeze_opt(int argc, VALUE *argv)
     static ID keyword_ids[1];
     VALUE opt;
     VALUE kwfreeze;
+    int ret = 1;
 
     if (!keyword_ids[0]) {
 	CONST_ID(keyword_ids[0], "freeze");
     }
     rb_scan_args(argc, argv, "0:", &opt);
     if (!NIL_P(opt)) {
 	rb_get_kwargs(opt, keyword_ids, 0, 1, &kwfreeze);
-	if (kwfreeze == Qfalse) return FALSE;
-	if (kwfreeze != Qundef && kwfreeze != Qtrue) {
-	    rb_raise(rb_eArgError, "unexpected value for freeze: %"PRIsVALUE,
-		     rb_obj_class(kwfreeze));
-	}
+    ret = obj_freeze_opt(kwfreeze);    
```

Comment:
Currently only `freeze` option is used and `kwfreeze` can't be `Qundef` here, but it would be safer to check for it.
And trailing spaces.